cpp/tests/utilities/cudf_test_utils.cuh

```diff
@@ -16,114 +16,69 @@
-#include "tuple_vectors.h"
-
-#include <cudf.h>
-#include <rmm/rmm.h>
-#include <cudf/functions.h>
-#include <utilities/cudf_utils.h>
-#include <utilities/bit_util.cuh>
-
 #include <bitset>
 #include <numeric> // for std::accumulate
 #include <memory>
-
 #include <thrust/equal.h>
 
+#include "gtest/gtest.h"
+
+// See this header for all of the recursive handling of tuples of vectors
+#include "tuple_vectors.h"
+#include "cudf.h"
+#include "rmm/rmm.h"
+#include "cudf/functions.h"
+#include "utilities/cudf_utils.h"
+#include "utilities/bit_util.cuh"
+#include "utilities/type_dispatcher.hpp"
+
+
 // Type for a unique_ptr to a gdf_column with a custom deleter
 // Custom deleter is defined at construction
 using gdf_col_pointer = typename std::unique_ptr<gdf_column, 
                                                  std::function<void(gdf_column*)>>;
 
-template <typename col_type>
-void print_typed_column(col_type * col_data, 
-                        gdf_valid_type * validity_mask, 
-                        const size_t num_rows)
-{
+namespace {
 
-  std::vector<col_type> h_data(num_rows);
-  cudaMemcpy(h_data.data(), col_data, num_rows * sizeof(col_type), cudaMemcpyDeviceToHost);
+struct column_printer {
+  template <typename ColumnType>
+  void operator()(gdf_column const* the_column) {
 
+    gdf_size_type const num_rows{the_column->size};
 
-  const size_t num_masks = gdf_get_num_chars_bitmask(num_rows);
-  std::vector<gdf_valid_type> h_mask(num_masks);
-  if(nullptr != validity_mask)
-  {
-    cudaMemcpy(h_mask.data(), validity_mask, num_masks * sizeof(gdf_valid_type), cudaMemcpyDeviceToHost);
-  }
+    ColumnType const* col_data{
+        static_cast<ColumnType const*>(the_column->data)};
 
+    std::vector<ColumnType> h_data(num_rows);
+    cudaMemcpy(h_data.data(), col_data, num_rows * sizeof(ColumnType),
+               cudaMemcpyDeviceToHost);
 
-  for(size_t i = 0; i < num_rows; ++i)
-  {
-    // If the element is valid, print it's value
-    if(true == gdf_is_valid(h_mask.data(), i))
-    {
-      std::cout << h_data[i] << " ";
+    gdf_size_type const num_masks{gdf_get_num_chars_bitmask(num_rows)};
+    std::vector<gdf_valid_type> h_mask(num_masks, ~gdf_valid_type{0});
+    if (nullptr != the_column->valid) {
+      cudaMemcpy(h_mask.data(), the_column->valid,
+                 num_masks * sizeof(gdf_valid_type), cudaMemcpyDeviceToHost);
     }
-    // Otherwise, print an @ to represent a null value
-    else
-    {
-      std::cout << "@" << " ";
+
+    for (gdf_size_type i = 0; i < num_rows; ++i) {
+      // If the element is valid, print it's value
+      if (true == gdf_is_valid(h_mask.data(), i)) {
+        std::cout << h_data[i] << " ";
+      }
+      // Otherwise, print an @ to represent a null value
+      else {
+        std::cout << "@"
+                  << " ";
+      }
     }
+    std::cout << std::endl;
   }
-  std::cout << std::endl;
+};
 }
 
 void print_gdf_column(gdf_column const * the_column)
```

Comment:
```suggestion
inline void print_gdf_column(gdf_column const * the_column)
```

Reply:
> add some `inline`. helpful when tests span multiple source files.

I beg to differ. While `inline` can be used to avoid instantiation conflicts between different translation units, that is relevant (almost) only in the case of small functions which it makes sense to inline. In this case, what I believe should happen is moving this function's definition into its own `.cu` or `.cpp` file.

Reply:
This function **IS** just one line now.

Reply:
@devavret : Technically you're right, but it actually "contains" the helper class, which gets instantiated for all dtypes. So I still think that should all go into a separate TU.

Reply:
It's a non-issue because the implementation will eventually just move into `column_wrapper<T>::print()`.